Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1202,6 +1206,20 @@ GridTrackSizingDirection RenderGrid::autoPlacementMinorAxisDirection() const
     return (autoPlacementMajorAxisDirection() == ForColumns) ? ForRows : ForColumns;
 }
 
+void RenderGrid::trimItemMarginsAdjacentToGridEdge(MarginTrimType marginTrimType)
+{
+    if (marginTrimType != MarginTrimType::BlockStart) {
```

Comment:
This statement is redundent. `setTrimmedMarginForChild` should handle this case.

Let's just remove it.

Reply:
After all the flexbox patches I think this is no longer redundant. I think the early return would have been useful in either case because the logic below only applies to the block-start margin  and would be incorrect for other values of `MarginTrimType` on non-debug builds